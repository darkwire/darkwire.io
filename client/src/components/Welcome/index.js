import React, { Component } from 'react'
import PropTypes from 'prop-types'
import RoomLink from 'components/RoomLink'

class Welcome extends Component {
  constructor(props) {
    super(props)
    this.state = {
      roomUrl: `https://darkwire.io/${props.roomId}`,
    }
  }

  render() {
    return (
      <div>
        <div>
          v2.0 is a complete rewrite and includes several new features. Here are some highlights:
          <ul className="native">
            <li>Support on all modern browsers (Chrome, Firefox, Safari, Safari iOS, Android)</li>
            <li>Slash commands (/nick, /me, /clear)</li>
            <li>Room owners can lock the room, preventing anyone else from joining</li>
            <li>Front-end rewritten in React.js and Redux</li>
            <li>Send files up to 4 MB</li>
          </ul>
          <div>
            You can learn more <a href="https://github.com/darkwire/darkwire.io" target="_blank" rel="noopener noreferrer">here</a>.
          </div>
        </div>
        <br />
        <p className='mb-2'>Others can join this room using the following URL:</p>
        <RoomLink roomId={this.props.roomId} translations={this.props.translations} />
        <div className="react-modal-footer">
          <button className="btn btn-primary btn-lg" onClick={this.props.close}>{this.props.translations.welcomeModalCTA}</button>
        </div>
      </div>
    )
  }
}

Welcome.propTypes = {
  roomId: PropTypes.string.isRequired,
  close: PropTypes.func.isRequired,
  translations: PropTypes.object.isRequired,
}

export default Welcome
