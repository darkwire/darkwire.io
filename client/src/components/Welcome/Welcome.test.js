import React from 'react';
import { render } from '@testing-library/react';

import Welcome from '.';

test('Welcome component is displaying', async () => {
  const { asFragment } = render(
    <Welcome roomId='roomtest' close={() => {}} translations={{}} />
  );

  expect(asFragment()).toMatchSnapshot();
});
