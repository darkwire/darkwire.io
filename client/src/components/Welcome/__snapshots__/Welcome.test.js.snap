// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component is displaying 1`] = `
<DocumentFragment>
  <div>
    <div>
      v2.0 is a complete rewrite and includes several new features. Here are some highlights:
      <ul
        class="native"
      >
        <li>
          Support on all modern browsers (Chrome, Firefox, Safari, Safari iOS, Android)
        </li>
        <li>
          Slash commands (/nick, /me, /clear)
        </li>
        <li>
          Room owners can lock the room, preventing anyone else from joining
        </li>
        <li>
          Front-end rewritten in React.js and Redux
        </li>
        <li>
          Send files up to 4 MB
        </li>
      </ul>
      <div>
        You can learn more 
        <a
          href="https://github.com/darkwire/darkwire.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          here
        </a>
        .
      </div>
    </div>
    <br />
    <p
      class="mb-2"
    >
      Others can join this room using the following URL:
    </p>
    <form>
      <div
        class="form-group"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="room-url"
            readonly=""
            type="text"
            value="http://localhost/roomtest"
          />
          <div
            class="input-group-append"
          >
            <button
              class="copy-room btn btn-secondary"
              data-clipboard-text="http://localhost/roomtest"
              data-placement="bottom"
              data-toggle="tooltip"
              type="button"
            >
              <svg
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </form>
    <div
      class="react-modal-footer"
    >
      <button
        class="btn btn-primary btn-lg"
      />
    </div>
  </div>
</DocumentFragment>
`;
